RNN.py

```diff
@@ -13,32 +13,38 @@
 import numpy as np
 from helper import *
 import csv
+import IPython
 
 class RNN(nn.Module):
     def __init__(self, input_size, num_filters, output_size,
-                 kernel_size, lstm_hidden, use_gpu, batch_size, n_layers=1):
+                 kernel_size_l1, lstm_hidden, use_gpu, batch_size, n_layers=1):
         super(RNN, self).__init__()
         self.input_size = input_size # Should just be 1.
         self.num_filters = num_filters
         self.output_size = output_size # Number of AAs
         self.n_layers = n_layers # Defaults to one.
 
-        self.kernel_size = kernel_size
+        self.kernel_size_l1 = kernel_size_l1
         self.lstm_hidden = lstm_hidden
         self.use_gpu = use_gpu
         self.batch_size = batch_size
 
-        p1 = kernel_size
-        self.c1 = nn.Conv1d(input_size, num_filters, kernel_size, padding=p1)
-        dilation = 1
-        p2 = kernel_size + (kernel_size - 1) * dilation
-        self.c2 = nn.Conv1d(input_size, num_filters, kernel_size,
-                            dilation=dilation,
-                            padding=p2)
+        self.convs = []
+        for i in xrange(0,len(kernel_size_l1)):
```

Comment:
you should also pass an array of dilations. and use those for each of the convolution elements.